wgpu-core/src/command/transfer.rs

```diff
@@ -363,6 +363,59 @@ pub(crate) fn validate_linear_texture_data(
     Ok((bytes_in_copy, image_stride_bytes))
 }
 
+/// Validation for texture/buffer copies.
+///
+/// This implements the following checks from WebGPU's [validating texture buffer copy][vtbc]
+/// algorithm:
+///  * The texture must not be multisampled.
+///  * The copy must be from/to a single aspect of the texture.
+///  * If `aligned` is true, the buffer offset must be aligned appropriately.
+///
+/// Other checks in this algorithm are enforced elsewhere.
+///
+/// [vtbc]: https://gpuweb.github.io/gpuweb/#abstract-opdef-validating-texture-buffer-copy
+pub(crate) fn validate_texture_buffer_copy<T>(
+    texture_copy_view: &wgt::TexelCopyTextureInfo<T>,
+    aspect: hal::FormatAspects,
+    desc: &wgt::TextureDescriptor<(), Vec<wgt::TextureFormat>>,
+    offset: BufferAddress,
+    aligned: bool,
+) -> Result<(), TransferError> {
+    if desc.sample_count != 1 {
+        return Err(TransferError::InvalidSampleCount {
+            sample_count: desc.sample_count,
+        });
+    }
+
+    if !aspect.is_one() {
+        return Err(TransferError::CopyAspectNotOne);
+    }
+
+    let mut offset_alignment = if desc.format.is_depth_stencil_format() {
+        4
+    } else {
+        desc.format
+            .block_copy_size(Some(texture_copy_view.aspect))
+            .unwrap_or(1)
```

Comment:
I think we need to error if we get back a `None` since we don't check anywhere else that the `format-aspect` combination is copyable.

`guaranteed_format_features` also seems to report that `Depth24Plus` is copyable which isn't right but that's a different (though related) issue.

Reply:
Let me know if you still think there are invalid copies that aren't getting flagged, or if you'd prefer to move the special checks for non-copyable depth/stencil into this function.

Reply:
Oh yeah, the last change looks good! I forgot we had `conv::is_valid_copy_{src,dst}_texture_format`.